### ISSUE TITLE
Why is default line gap 200?

### ISSUE THREAD

**Issue by jamesgk** [googlei18n/ufo2ft#76]
Is 0 a better default value? I think it's used in Glyphs.

**Comment by dougfelt** [comment]
Noto lint expects this to be zero.

**Comment by marekjez86** [comment]
copied from an e-mail:   "Unfortunately we have 3 ascenders, 3 descenders and 2 line gaps. To get
predictable results we give the same values to ascenders and descenders
and leave the line gaps at 0.  The ascender, descender, cap-height,
x-height values are stored in the masters. These should be used, unless
overwritten by the instances with custom parameters. Linegap should be 0
unless defined for hhea or OS/2 in instance custom parameters."

**Comment by moyogo** [comment]
The problem here is that that ascender and descender when similar to cap-height and x-height, meaning when they represent actual ascender approximation value and descender approximation value, are not enough to define the lineheight.

The line height is usually the sum of absolute values of ascender, descender and line gap.
If a master has no line gap information, it doesn’t have clear line height information.

One can only guess whether the ascender and descender values represent the approximate yMax of letters with ascender and yMin of letters with descender or if they represent what one wants to go in hhea.ascender and hhea.descender, os2.typoAscender and os2.typoDescender, or even os2.winAscent and os2.winDescent.

To be honest, source font formats should have a lineheight or linegap field to resolve this ambiguity.

ufo2fdk and hence ufo2ft make the assumption that the lineheight is 1200 (when unitsPerEm is 1000), because it assumes that the sum of absolute value of ascender and descender is equal ot unitsPerEm. See the recommendations for 'OS/2' table in https://www.microsoft.com/typography/otspec/recom.htm.
Since os2.typoAscender and os2typoDescender are meant to represent the "real" ascender and descender, the same recommendation can be applied to ascender and descender. Hence the recommandation for linegap being 200.

**Comment by davelab6** [comment]
On 28 October 2016 at 16:37, Denis Moyogo Jacquerye <
notifications@github.com> wrote:

> Hence the recommandation for linegap being 200.


Kalapi and Khaled researched this for GF a few months ago, I was meant to
write a concise summary in
https://github.com/googlefonts/gf-docs/blob/master/ProjectChecklist.md#vertical-metrics
but didn't yet do so. The details are all in
https://groups.google.com/d/topic/googlefonts-discuss/W4PHxnLk3JY/discussion

**Comment by moyogo** [comment]
>  Hence the recommandation for linegap being 200.

Let me clarify that: the *OT spec* recommendation *for Western fonts*.

There is no issue with having hhea.lineGap=0, it might actually be better. Some implementations use hhea.lineGap when they calculate the lineheight but others don’t, so a common practice is to set it to 0 and pad hhea.ascender and hhea.descender so the absolute sum of the three adds up to the correct lineheight.

For os2.typoLineGap, implementations do use it properly, so it doesn’t have to be zero, but can be if you just want to have the os2.typo metrics math to the hhea metrics.

To sum up, a hhea.lineGap=0 only makes sense if the hhea.ascender and hhea.descender aren’t both equal to the real ascender and descender. And a os2.typoLineGap=0 only makes sense if the os2.typoAscender and os2.typoDescender aren’t both equal to the real ones either.
